### PR TITLE
Set USE_KINETO=1

### PR DESCRIPTION
**Pull request by ilia-cher**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#49897 Set USE_KINETO=1**

Summary:
Resend of https://github.com/pytorch/pytorch/pull/49201

Test Plan:
see 49201

Differential Revision: [D25717102](https://our.internmc.facebook.com/intern/diff/D25717102)

### CODE REVIEW

**Comment by ilia-cher** [comment]
I reverted earlier 49201 because of some ddp test failures in one of the master builds, will investigate

**Review by malfet — APPROVED** [review]

**Comment by malfet** [comment]
@ilia-cher please make sure to check that your PR does not add any direct dependencies against libcuda.so (i.e. PyTorch build with CUDA should be loadable on systems that do not have CUDA driver installed)

**Comment by ilia-cher** [comment]
> @ilia-cher please make sure to check that your PR does not add any direct dependencies against libcuda.so (i.e. PyTorch build with CUDA should be loadable on systems that do not have CUDA driver installed)

yes, I think we should be fine - we only compile cupti and kineto statically, and CI has non-CUDA builds as well

**Comment by ilia-cher** [comment]
CI looks good (mod. two unrelated failures)

File: cmake/Dependencies.cmake [hunk 1]
```diff
@@ -1787,43 +1787,68 @@ list(APPEND Caffe2_DEPENDENCY_LIBS fmt::fmt-header-only)
...
+  set(KINETO_LIBRARY_TYPE "static" CACHE STRING "")
+  set(CUDA_SOURCE_DIR "${CUDA_TOOLKIT_ROOT_DIR}" CACHE STRING "")
+
+  message(STATUS "Configuring Kineto dependency:")
+  message(STATUS "  KINETO_SOURCE_DIR = ${KINETO_SOURCE_DIR}")
+  message(STATUS "  KINETO_BUILD_TESTS = ${KINETO_BUILD_TESTS}")
+  message(STATUS "  KINETO_LIBRARY_TYPE = ${KINETO_LIBRARY_TYPE}")
+  message(STATUS "  CUDA_SOURCE_DIR = ${CUDA_SOURCE_DIR}")
+
+  if(EXISTS ${CUDA_SOURCE_DIR}/extras/CUPTI/lib64/libcupti_static.a)
```
Comment (ngimel) [inline cmake/Dependencies.cmake:1815]: Is it possible to use CUDA_cupti_LIBRARY found by FindCUDA here? Also, see #51026, USE_CUPTI_SO should be given preference.

**Pull request closed without merging**